phpBB/phpbb/plupload/plupload.php

```diff
@@ -263,11 +263,24 @@ public function generate_resize_string()
 		$resize = '';
 		if ($this->config['img_max_height'] > 0 && $this->config['img_max_width'] > 0)
 		{
-			$resize = sprintf(
-				'resize: {width: %d, height: %d, quality: 85},',
-				(int) $this->config['img_max_width'],
-				(int) $this->config['img_max_height']
-			);
+			if ($this->config['img_strip_metadata'] == 1)
+			{
+				$resize = sprintf(
+					'resize: {width: %d, height: %d, quality: %d, preserve_headers: false},',
+					(int) $this->config['img_max_width'],
+					(int) $this->config['img_max_height'],
+					(int) $this->config['img_quality']
+				);
+			}
+			else
+			{
+				$resize = sprintf(
+					'resize: {width: %d, height: %d, quality: %d},',
+					(int) $this->config['img_max_width'],
+					(int) $this->config['img_max_height'],
+					(int) $this->config['img_quality']
+				);
+			}
```

Comment:
Doesn't work this way?

```
$resize = sprintf(
	'resize: {width: %d, height: %d, quality: %d, preserve_headers: %s},',
	(int) $this->config['img_max_width'],
	(int) $this->config['img_max_height'],
	(int) $this->config['img_quality'],
	(bool) $this->config['img_strip_metadata']
);
```

Reply:
I looked at whether I could simplify this, so thanks for the suggestion.

I'd also need to invert the value, or invert the config setting (eg from strip metadata to preserve metadata). But in any event, it doesn't work. It is ok when preserve_headers: true, but when false the attachment function fails. After uploading the file, there is an extra empty file upload line and the image cannot be placed inline (see below).

![failed upload](https://user-images.githubusercontent.com/58344533/72700649-904d1b80-3ba0-11ea-8fa9-7c7f18fbf846.PNG)

Reply:
Try something like, no need to compare the config since it is true if 1 false otherwise.

```
$my_var = $this->config['img_strip_metadata'] ? 'false' : 'true';

$resize = sprintf(
	'resize: {width: %d, height: %d, quality: %d, preserve_headers: %s},',
	(int) $this->config['img_max_width'],
	(int) $this->config['img_max_height'],
	(int) $this->config['img_quality'],
	$my_var
);
```

Reply:
Yep, this way did it. Thanks very much for the tip.  I have amended and will push the change when I fix the plupload test.